.infrastructure/helm-chart/todoapp/templates/app/deployment.yml

```diff
@@ -0,0 +1,92 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .Chart.Name }}-deployment
+  namespace: {{ .Values.common.namespace }}
+spec:
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxSurge: {{ .Values.deployment.rollingUpdate.maxSurge }}
+      maxUnavailable: {{ .Values.deployment.rollingUpdate.maxUnavailable }}
```

Comment:
Ensure that `.Values.deployment.rollingUpdate.maxSurge` and `.Values.deployment.rollingUpdate.maxUnavailable` are defined in your `values.yaml` file for proper rolling update strategy configuration.